tests/test-menubar.lua

```diff
@@ -37,22 +37,33 @@ do
     end
 end
 
+local show_menubar_and_hide = function(count)
+    -- Just show the menubar and hide it.
+    -- TODO: Write a proper test. But for the mean time this is better than
+    -- nothing (and tells us when errors are thrown).
+
+    if count == 1 then
+        menubar.show()
+    end
+
+    -- Test that the async population of the menubar is done
+    if menubar_refreshed then
+        menubar.hide()
+        awesome.sync()
+        return true
+    end
+end
+
 runner.run_steps {
     function(count)
-        -- Just show the menubar and hide it.
-        -- TODO: Write a proper test. But for the mean time this is better than
-        -- nothing (and tells us when errors are thrown).
-
-        if count == 1 then
-            menubar.show()
-        end
+        -- Show and hide with defaults
+        show_menubar_and_hide(count)
```

Comment:
you need to `return true` when the test passed, otherwise it retries and timeout. You can run these in your computer by running `tests/run.sh tests/test-menubar.lua`